scikitplot/plotters.py

```diff
@@ -87,6 +94,50 @@ def plot_confusion_matrix(y_true, y_pred, labels=None, title=None, normalize=Fal
     else:
         classes = np.asarray(labels)
 
+    def validate_labels(known_classes, passed_labels, argument_name):
+        known_classes = np.array(known_classes)
+        passed_labels = np.array(passed_labels)
+
+        unique_labels, unique_indexes = np.unique(passed_labels, return_index=True)
+
+        if len(passed_labels) != len(unique_labels):
+            indexes = np.arange(0, len(passed_labels))
+            duplicate_indexes = indexes[~np.isin(indexes, unique_indexes)]
+            duplicate_labels = passed_labels[duplicate_indexes]
+
+            msg = "The following duplicate labels were passed into %s: %s" % (argument_name, ", ".join(duplicate_labels))
+            raise ValueError(msg)
+
+        passed_labels_absent = ~np.isin(passed_labels, known_classes)
+
+        if np.any(passed_labels_absent):
+            absent_labels = passed_labels[passed_labels_absent]
+
+            msg = "The following labels were passed into %s, but were not found in labels: %s" % (argument_name, ", ".join(absent_labels))
+            raise ValueError(msg)
+
+        return
+
+    if true_labels is None:
+        true_classes = classes
+    else:
+        validate_labels(classes, true_labels, "true_labels")
+
+        true_label_indexes = np.where(np.isin(classes, true_labels))
+
+        true_classes = classes[true_label_indexes]
+        cm = cm[true_label_indexes]
+
+    if pred_labels is None:
+        pred_classes = classes
+    else:
+        validate_labels(classes, pred_labels, "pred_labels")
+
+        pred_label_indexes = np.where(np.isin(classes, pred_labels))
+
+        pred_classes = classes[pred_label_indexes]
+        cm = cm[:,pred_label_indexes][:,0,:]
+
     if normalize:
```

Comment:
I think you should put this step *before* the validation, otherwise the calculated normalized values might be wrong